### PR TITLE
add zeepr project

### PR DESCRIPTION
**Pull request by JoJo-Joker**
Name (to be shown on DefiLlama):
Zeepr

Twitter Link:
https://twitter.com/Zeepr_labs

List of audit links if any:
N/A

Website Link:
https://www.zeepr.io/

Current TVL:
2.11B

Treasury Addresses (if the protocol has treasury)
N/A

Chain:
Arbitrum、BNB Chain、Polygon、EOS EVM、Core

Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
N/A

Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
N/A

Short Description (to be shown on DefiLlama):
Zeepr is the world's first decentralized perpetual contract protocol that supports both Coin-margin and U-margin, amplifying traders' profits.

Token address and ticker if any:
0xe46C5eA6Da584507eAF8dB2F3F57d7F578192e13

Category (full list at https://defillama.com/categories) *Please choose only one:
4.Dexes

forkedFrom (Does your project originate from another project):
No

Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/Zeeprlabs

Logo (High resolution, will be shown with rounded borders):
photo_6336565470735481090_x

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/zeepr exports TVL: 
        
 
 
    arbitrum                  0
    
    total                    0

**Comment by JoJo-Joker** [comment]
hi, team, may i ask why this total is 0?

**Comment by g1nt0ki** [comment]
@JoJo-Joker because the token is not priced, can you get it listed on coingecko first?

**Comment by JoJo-Joker** [comment]
@g1nt0ki can i count my own token as Staking?

**Comment by JoJo-Joker** [comment]
@g1nt0ki May I ask if it is possible to launch on gate or coinmarketcap? Or can only be listed on Coinsecko?

**Comment by g1nt0ki** [comment]
Atm we rely only on coingecko for pulling token prices

**Comment by JoJo-Joker** [comment]
@g1nt0ki We have launched on GeckoTerminal, which seems to belong to Coinsecko. Can we use it?

**Comment by g1nt0ki** [comment]
not, it is not part of the coingecko api that we have access to, bigger issue is, we cant count your own token as tvl

**Comment by JoJo-Joker** [comment]
@g1nt0ki  May I ask what can my own token count as? Can it be considered as stacking?

**Comment by g1nt0ki** [comment]
yeah, under staking

**Comment by g1nt0ki** [comment]
https://docs.llama.fi/list-your-project/what-to-include-as-tvl

**Comment by JoJo-Joker** [comment]
@g1nt0ki  May I ask if I still need to be listed on Coinsecko after replacing it with stacking?

**Comment by g1nt0ki** [comment]
yes, to price the token

**Comment by JoJo-Joker** [comment]
@g1nt0ki  May I choose not to display TVL and instead display other data? such as stacking or others

**Comment by g1nt0ki** [comment]
yes, we can show staking and 0 tvl

**Comment by llamatester** [comment]
Error while running adapter at projects/zeepr: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:11
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:138:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
Error while running adapter at projects/zeepr: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:11
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:138:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
Error while running adapter at projects/zeepr: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:11
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:138:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by JoJo-Joker** [comment]
@g1nt0ki  May I write it this way?

File: projects/zeepr/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const { staking } = require("./helper/staking");
+
+module.exports = {
+  arbitrum: {
+    tvl: 0,
+    staking: staking("0xe46C5eA6Da584507eAF8dB2F3F57d7F578192e13"),
```
Comment (g1nt0ki) [inline projects/zeepr/index.js:6]: no, this doesnt work, how do you expect us compute tvl, where staking contract is not provided, also, we still need a way to price the token

Comment (JoJo-Joker) [reply projects/zeepr/index.js:6]: If I only display staking, how should I write it? @g1nt0ki

Comment (g1nt0ki) [reply projects/zeepr/index.js:6]: `staking(stakingAddress, tokenAddress)`

**Comment by llamatester** [comment]
Error while running adapter at projects/zeepr: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:11
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:138:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
Error while running adapter at projects/zeepr: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:11
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:138:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
The adapter at projects/zeepr exports TVL: 
        
 
 
    arbitrum                  0
    arbitrum-staking          0
    staking                   0
    
    total                    0

**Comment by g1nt0ki** [comment]
you did not put a smart contract but a wallet for staking?

**Comment by JoJo-Joker** [comment]
@g1nt0ki I have set a totalMargins smart contract method on the address 0xe46C5eA6Da584507eAF8dB2F3F57d7F578192e13. By passing in the address parameter, the corresponding balance can be queried. Is this type of smart contract acceptable?

**Comment by llamatester** [comment]
The adapter at projects/zeepr exports TVL: 
        
 
 
    arbitrum                  0
    
    total                    0

**Comment by JoJo-Joker** [comment]
@g1nt0ki  Hello, I added price to Coinsecko. Is it okay to write the code like this? Can tvl be calculated like this?

**Comment by g1nt0ki** [comment]
sorry, I am tired of repeating myself please read my previous comments

**Comment by llamatester** [comment]
The adapter at projects/zeepr exports TVL: 
        
 
 
    arbitrum                  0
    arbitrum-staking          0
    staking                   0
    
    total                    0

**Comment by JoJo-Joker** [comment]
@g1nt0ki Hello, sorry to bother you again. May I ask if my staking is in a wallet not a smart contract, can it be displayed? If possible, which method should I use?

**Comment by g1nt0ki** [comment]
@JoJo-Joker sorry, we cant count value in EOA as tvl

**Comment by JoJo-Joker** [comment]
@g1nt0ki ok, I see, thank you for your reply, and may I ask if the value in EOA cannot be used as tvl? What can this value be used for? such as pool2 or others?

**Pull request closed without merging**